src/components/Controls/Controls.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+export default function Controls({ setContinent, countries, setSearchValue }) {
+  const continents = [...new Set(countries.map(({ continent }) => continent))];
+
+  const handleChange = (event) => {
+    setSearchValue(event.target.value);
+  };
+
+  return (
+    <div className="controls">
+      <input type="text" placeholder="Search by name" onChange={handleChange}></input>
+      <select onChange={(event) => setContinent(event.target.value)}>
+        <option value="all">All</option>
+        {continents.map((continent) => (
+          <option key={continent} value={continent}>
```

Comment:
you might want to include a check for ```null``` as currently there is a blank option in the dropdown that when selected displays no flags